internal/api/testhelpers.go

```diff
@@ -140,3 +143,30 @@ func Must[T any](v T, err error) T {
 	}
 	return v
 }
+
+// Helper function to create a subscription with AllowDevNonStableChannels feature flag
+func CreateSubscriptionWithNonStableChannels() *arm.Subscription {
+	return &arm.Subscription{
+		State:            arm.SubscriptionStateRegistered,
+		RegistrationDate: ptr.To(time.Now().Format(time.RFC1123)),
+		Properties: &arm.SubscriptionProperties{
+			RegisteredFeatures: &[]arm.Feature{
+				{
+					Name:  ptr.To(FeatureAllowDevNonStableChannels),
+					State: ptr.To("Registered"),
+				},
+			},
+		},
+	}
+}
+
+// Helper function to create a subscription without AllowDevNonStableChannels feature flag
+func CreateStandardSubscription() *arm.Subscription {
+	return &arm.Subscription{
+		State:            arm.SubscriptionStateRegistered,
+		RegistrationDate: ptr.To(time.Now().Format(time.RFC1123)),
+		Properties: &arm.SubscriptionProperties{
+			RegisteredFeatures: &[]arm.Feature{},
+		},
+	}
+}
```

Comment:
How about if we combine these and take registered features as a variadic argument list?
```
func CreateTestSubscription(registeredFeatures ...string)
```
That will help future-proof it if we add more subscription features.